src/com/amazon/ion/impl/lite/IonStructLite.java

```diff
@@ -373,10 +378,13 @@ private int find_field_helper(String fieldName)
         if (isNullValue()) {
             // nothing to see here, move along
         }
-        else if (_field_map != null) {
+        else if (_child_count > STRUCT_INITIAL_SIZE || _field_map != null) {
```

Comment:
>In some places we use _child_count directly, and in others where we could we instead use get_child_count(). What governs the choice?

In the new code in this PR, I use `_child_count` directly because there is no need to use an accessor method for a `protected` variable from an ancestor. I did not care to change existing usages (30 of them) and further bloat this PR. Here's an issue for standardizing them: #560

Reply:
>private boolean fieldMapIsActive(int proposedSize) {
    if (_field_map != null) return true;
    if (proposedSize <= STRUCT_INITIAL_SIZE) return false;
    build_field_map();
    return true;
}

Done, thank you for the suggestion.

Regarding `transitionToLargeSize`, I agree that it should be overhauled/eliminated. The `nextSize` thing is too wonky for me to want to touch it as part of this effort, though, as I'd like to spend a lot more time focusing on the performance of the related IonStruct operations. Right now, `nextSize` is called when children are added. I think we could change this and make field map creation fully lazy, but again, I want to have the time set aside to measure the performance implications for existing use cases. #561